mitodl/open-discussions
# Added channel description to API and UI

#### What are the relevant tickets?
Part of #263 

#### What's this PR do?
Adds the channel description to the API and renders it as markdown on the channel page.

#### How should this be manually tested?
- Go to /api/v0/channels/:channel_name/ and verify you see the description field returned.
- Update the channel description with some markdown like (not all these markdown examples will work, I was lazy and copy/pasted an excessive example):
```python
from django.contrib.auth.models import User
from channels.api import Api
user = User.objects.get(username="mitodl")
api = Api(user)
api.update_channel("channel_name", description="""
An h1 header
============

Paragraphs are separated by a blank line.

2nd paragraph. *Italic*, **bold**, and `monospace`. Itemized lists
look like:

  * this one
  * that one
  * the other one

Note that --- not considering the asterisk --- the actual text
content starts at 4-columns in.

> Block quotes are
> written like so.
>
> They can span multiple paragraphs,
> if you like.

Use 3 dashes for an em-dash. Use 2 dashes for ranges (ex., "it's all
in chapters 12--14"). Three dots ... will be converted to an ellipsis.
Unicode is supported. ☺

An h2 header
------------

Here's a numbered list:

 1. first item
 2. second item
 3. third item

### An h3 header ###

Now a nested list:

 1. First, get these ingredients:

      * carrots
      * celery
      * lentils
2. hiiiii

Notice again how text always lines up on 4-space indents (including
that last line which continues item 3 above).

Here's a link to [a website](http://foo.bar), to a [local
doc](local-doc.html), and to a [section heading in the current
doc](#an-h2-header). 

A horizontal rule follows.

***

Here's a definition list:

apples
  : Good for making applesauce.
oranges
  : Citrus!
tomatoes
  : There's no "e" in tomatoe.

Again, text is indented 4 spaces. (Put a blank line between each
term/definition pair to spread things out more.)

Here's a "line block":

| Line one
|   Line too
| Line tree

""")
````
- Verify this content and/or the default value is correctly rendered in the sidebar.
- Ensure no adverse affects on other pages in the app (e.g. pages can load and layout is unaffected by changes to support right sidebar).

#### Screenshots

![localhost_8063_channel_odl](https://user-images.githubusercontent.com/28598/33038565-164b7a82-ce03-11e7-9514-e0a86734cf2b.png)
